### PR TITLE
[RFC] Attempt to implement Google OAuth including getting profile email via PKCE

### PR DESCRIPTION
**Pull request by clarkg**

### CODE REVIEW

**Comment by clarkg** [comment]
@scotttrinh (idk why I can't assign somebody.  thanks in advance!)

**Comment by scotttrinh** [comment]
Thanks for the PR! I don't think we'll probably merge something that is so provider-specific, but having some general functionality to demonstrate and figuring out the minimal amount of supporting utility functions, so I'll do some review on this and try help guide us toward something that will be useful for someone using any of our supported providers.

**Comment by clarkg** [comment]
@scotttrinh yeah thanks! I'm happy to contribute and merge as a side-effect, but as you know my motivation anyway is asking for help debugging this from our convo on https://discord.com/channels/841451783728529451/1260648055737290773

File: app/auth/[...auth]/route.ts [hunk 1]
```diff
@@ -14,6 +17,38 @@ export const { GET, POST } = auth.createAuthRouteHandlers({
...
+    });
+
+    if (codeExchangeResponse.status !== 200) {
+      console.error("Code exchange failed: ", codeExchangeResponse.status, codeExchangeResponse.statusText);
+      redirect("/error?error=code-exchange-failed");
+    }
+
+    const { auth_token, provider_token } = await codeExchangeResponse.json();
+
+    const profile = await getUserProfile(provider_token);
```
Comment (scotttrinh) [inline app/auth/[...auth]/route.ts:50]: The `provider_token` is already (potentially) in the `tokenData`, so this is all you need:

```suggestion
    const { provider_token } = tokenData;
    const profile = provider_token && await getUserProfile(provider_token);
```

**Comment by clarkg** [comment]
@scotttrinh thank you, that worked!  will close this PR but lmk if you'd like me to re-open and contribute this if helpful

**Comment by scotttrinh** [comment]
@clarkg 

Maybe we can have a function defined in this module that, given a `tokenData` will get some basic `profile` information and just comment that function heavily so that if someone was reading through the code, they could see how to get this kind of functionality?

Feel free to take a swing at add that, or I'm happy to add it to my todo list. Let me know either way. Thanks for jumping in here to contribute!

**Pull request closed without merging**